### PR TITLE
Add tfidf script

### PR DESCRIPTION
**Pull request by lgalke**
Usage: `index_tfidf -d txt/ -o tmp/tfidf`

### CODE REVIEW

File: scripts/index_tfidf [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+                        help="Path to txt file root")
+    parser.add_argument("-o", "--outpath", required=True,
+                        help="Path to output directory")
+    parser.add_argument("-n", "--n-features", default=None, type=int,
+                        help="Use that many features (default unlimited)")
+    parser.add_argument("--hash", action='store_true', default=False,
+                        help="Use (stateless) hashing vectorizer")
+
+if __name__ == "__main__":
+    import argparse
```
Comment (beduffy) [inline scripts/index_tfidf]: Is there any reason you import stuff stuff other than at the top of the file? 
I kinda like the look and feel though

File: aidistillery/index_tfidf.py [hunk 2]
```diff
@@ -70,9 +58,10 @@ def _add_args(parser):
     parser.add_argument("--hash", action='store_true', default=False,
                         help="Use (stateless) hashing vectorizer")
 
+
 if __name__ == "__main__":
```
Comment (beduffy) [inline aidistillery/index_tfidf.py]: Keeping this here unlike what we did in the other subcommands. 
I argue that we can keep both the ability to run the script directly AND the subcommand. This is more user friendly and sticks with the easy of having 1 script being able to do 1 thing with the script alone. Distill subcommand ties everything together.
Do you agree @lgalke @vinid ?

File: scripts/distill [hunk 3]
```diff
@@ -11,6 +11,7 @@ from aidistillery.fasttext_embeddings import fasttext_add_args, fasttext_main
 from aidistillery.document_embeddings import lsa_main, lsa_add_args
 from aidistillery.document_embeddings import doc2vec_main, doc2vec_add_args
 from aidistillery.named_entities import ner_main, ner_add_args
+from aidistillery.index_tfidf import tfidf_main, tfidf_add_args
 
 # A mapping from subcommand to main executable of subcommand
 class DistillSubcommand(object):
```
Comment (beduffy) [inline scripts/distill]: in the future we'll change Distill to Distil. US vs English spelling @lgalke :P

**Comment by beduffy** [comment]
I'm ready to merge if you are happy with my comments above @lgalke

File: scripts/index_tfidf (hunk 1 above)
Comment (lgalke) [reply scripts/index_tfidf]: At rare cases, it makes sense to do this. For this instance, argparse is only needed when the script is executed itself. When another part would import some function of it, argparse does not need to be imported. Other scenarios would be: you use matplotlib in one plot function but nowhere else in the module; then you do not know if the client will use the plotting function and you avoid one dependency as long as it is not really needed.

**Pull request merged**